qt/scientific_interfaces/test/ISISReflectometry/Batch/BatchPresenterTest.h

```diff
@@ -54,16 +54,42 @@ class BatchPresenterTest : public CxxTest::TestSuite {
     verifyAndClear();
   }
 
-  void testChildPresentersUpdatedWhenInstrumentChanged() {
+  void testInitInstrumentListUpdatesRunsPresenter() {
+    auto presenter = makePresenter();
+    EXPECT_CALL(*m_runsPresenter, initInstrumentList()).Times(1);
+    presenter.initInstrumentList();
+    verifyAndClear();
+  }
+
+  void testMainPresenterUpdatedWhenChangeInstrumentRequested() {
     auto presenter = makePresenter();
     auto const instrument = std::string("POLREF");
-    EXPECT_CALL(*m_runsPresenter, instrumentChanged(instrument)).Times(1);
-    EXPECT_CALL(*m_experimentPresenter, instrumentChanged(instrument)).Times(1);
-    EXPECT_CALL(*m_instrumentPresenter, instrumentChanged(instrument)).Times(1);
+    EXPECT_CALL(m_mainPresenter, notifyChangeInstrumentRequested(instrument))
+        .Times(1);
+    presenter.notifyChangeInstrumentRequested(instrument);
```

Comment:
I might misunderstand the mechanism behind this: but does calling `notifyChangeInstrumentRequested`not implicitly satisfy the mock expectation that it's called, making it always pass?

Reply:
Ah, I think that's been left in by mistake - I'll remove it.